packages/core/src/index.ts

```diff
@@ -92,8 +88,31 @@ export default class Core {
         config.onModelUpdate?.(this.#model);
       });
     }
+  }
 
-    this.#prepareUI();
+  /**
+   * Initialize and injects Plugin into the container
+   * @param plugin - allows to pass any implementation of editor plugins
+   */
+  public use(plugin: EditorjsPluginConstructor): void {
+    const eventBus = this.#iocContainer.get(EventBus);
+    const pluginType = plugin.type;
+    const api = this.#iocContainer.get(EditorAPI);
+
+    const pluginInstance = new plugin({
+      config: this.#config,
+      api,
+      eventBus,
+    });
+
+    this.#iocContainer.set(pluginType, pluginInstance);
```

Comment:
Might be useful here

https://docs.typestack.community/typedi/develop#using-service-groups